challenge.rb

```diff
@@ -1,26 +1,34 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  temp = []
+  input.each do |element|
+  	temp.push( element.capitalize )
+  end
+  return temp
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  temp = [input.detect { |x| x == "dog"}]
+  return temp
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  temp = input.reject {|x| x=="dog"}
+  return temp
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  temp = input.first(2)
+  return temp
```

Comment:
I agree on implicit returns, but the way the tests are set up, the input was altered from the tests before. The temp variable solved that problem.